pkg/agent/controller/policy/policy_controller.go

```diff
@@ -335,23 +336,23 @@ func (r *Reconciler) completePolicy(policy *securityv1alpha1.SecurityPolicy) ([]
 				DstIPBlocks:     policycache.DeepCopyMap(appliedIPBlocks).(map[string]*policycache.IPBlockItem),
 			}
 
+			ingressRuleTmpl.Ports, err = FlattenPorts(rule.Ports)
+			if err != nil {
+				return nil, err
+			}
+
 			if len(rule.From) == 0 {
+				ingressRule := ingressRuleTmpl.Clone()
 				// If "rule.From" is empty or missing, this rule matches all sources
 				ingressRule.SrcIPBlocks = map[string]*policycache.IPBlockItem{"": nil}
+				completeRules = append(completeRules, ingressRule)
 			} else {
-				// use policy namespace as ingress endpoint namespace
-				ingressRule.SrcGroups, ingressRule.SrcIPBlocks, err = r.getPeersGroupsAndIPBlocks(policy.Namespace, rule.From)
+				ingressRules, err := r.getCompleteRulesByParseSymmetricMode(ingressRuleTmpl, policy, networkingv1.PolicyTypeIngress, rule.From)
```

Comment:
networkingv1.PolicyTypeIngress is defined & used in NetworkPolicy. Although it has same str with ours, we'd better create a new const value?

Reply:
securityPolicy use networkingv1.PolicyType defines it's policy type. So I use it.